dynet/gpu-ops.h

```diff
@@ -34,6 +34,7 @@ void sparse_to_dense_assign(int n, const unsigned int* ids, float* src, float* t
 void dense_to_sparse_subtract(int n, const unsigned int* ids, float* src, float* trg);
 void sparse_to_dense_block_assign_and_multiply(int n, const unsigned *idx, int bsize, float mult, float *src, float *trg);
 void dense_to_sparse_block_add(int n, const unsigned* ids, int bsize, float* src, float* trg);
+void pad_input(float* output, const float* input, int N, int C, int H, int W, int pad_right, int pad_bottom);
```

Comment:
I think there may be an Eigen tensor operation "tensor.pad()" for this. Can you try to use this instead of writing a custom kernel? You can find an example in `eigen/unsupported/tests/cxx11_tensor_padding.cpp`.